File: src/main/java/com/homecraft/api/controller/AuthController.java
package com.homecraft.api.controller;

import com.homecraft.api.dto.LoginDTO;
import com.homecraft.api.entity.AuthResult;
import com.homecraft.api.security.JwtUtil;
import com.homecraft.api.service.AuthService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;
    private final JwtUtil jwtUtil;

    public AuthController(AuthService authService, JwtUtil jwtUtil) {
        this.authService = authService;
        this.jwtUtil = jwtUtil;
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody LoginDTO dto) {

        AuthResult result = authService.login(dto.getEmail(), dto.getPassword());

        if (!result.isSuccess()) {
            return ResponseEntity
                    .status(HttpStatus.UNAUTHORIZED)
                    .body(Map.of("message", result.getErrorMessage()));
        }

        String token = jwtUtil.generateToken(
                result.getUserId(),
                result.getRole(),
                result.getEmail()
        );

        return ResponseEntity.ok(
                Map.of(
                        "token", token,
                        "role", result.getRole()
                )
        );
    }
}

File: HomeCraft.API/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;

[ApiController]
[Route("api/auth")]
public class AuthController : ControllerBase
{
    private readonly IConfiguration _config;
    private readonly IAuthService _authService;

    public AuthController(IConfiguration config, IAuthService authService)
    {
        _config = config;
        _authService = authService;
    }

    [HttpPost("login")]
    public IActionResult Login(LoginDTO dto)
    {
        var result = _authService.Login(dto.Email, dto.Password);

        if (!result.Success)
            return Unauthorized(new { message = result.ErrorMessage });

        var token = JwtHelper.GenerateToken(
            result.UserId,
            result.Role,
            result.Email,
            _config);

        return Ok(new { token, role = result.Role });
    }
}
